x/oracle/keeper/grpc_query.go

```diff
@@ -61,12 +61,16 @@ func (q Keeper) Task(c context.Context, req *types.QueryTaskRequest) (*types.Que
 	}
 	ctx := sdk.UnwrapSDKContext(c)
 
-	task, err := q.GetTask(ctx, req.Contract, req.Function)
+	task, err := q.GetTask(ctx, types.NewTaskID(req.Contract, req.Function))
 	if err != nil {
 		return nil, err
 	}
-
-	return &types.QueryTaskResponse{Task: task}, nil
+	resp := types.QueryTaskResponse{}
```

Comment:
How to change this way
```
	if smartContractTask, ok := task.(*types.Task); ok {
		return types.QueryTaskResponse{Task: *smartContractTask}, nil
	}
	return nil, types.ErrCodeExists
```
```
// errors.go
ErrCastToTask = sdkerrors.Register(bankTypes.ModuleName, 9, "failed to cast to concrete task")
```

Reply:
Yeah, that looks good, will make the change on next commit.